odoo/sql_db.py

```diff
@@ -345,6 +345,10 @@ def execute(self, query, params=None, log_exceptions=True):
         try:
             params = params or None
             res = self._obj.execute(query, params)
+        except psycopg2.errors.ProgramLimitExceeded as e:
+            _logger.error("ProgramLimitExceeded: Index value too large.\nQuery: %s\nParams: %s\nError: %s", query, params, str(e))
+            raise odoo.exceptions.UserError("It looks like you're trying to import too many records at once or the data is too large. \n"
+                                            "Please try importing fewer records at a time or reduce the size of your input.")
```

Comment:
Error message should like 
```
"It looks like you're trying to process too many records at once or the data size is too large.\n"
    "Please try working with fewer records at a time or reduce the size of your input."
```
We can't raise UserError here. Instead, you raise a validation error (not sure we raise error in base)